Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -2497,3 +2500,16 @@ bool CLuaPedDefs::killPedTask(CClientPed* ped, taskType taskType, std::uint8_t t
             return false; 
     }
 }
+
+bool CLuaPedDefs::PlayPedVoiceLine(CClientPed* ped, int speechId, std::optional<float> probabilty)
+{
+    auto speechContextId = static_cast<ePedSpeechContext>(speechId);
+    if (speechContextId < ePedSpeechContext::NOTHING || speechContextId >= ePedSpeechContext::NUM_PED_CONTEXT)
+        throw LuaFunctionError("The argument speechId is invalid. The valid range is 0-359.");
+
+    if (probabilty.has_value() && probabilty < 0.0f)
+        return false;
+
+    ped->Say(speechContextId, probabilty.value_or(1.0f));
+    return true;
```

Comment:
You almost had it with the last change. The second check must also throw for negative values. And then you should realize that the function doesn't need a return value at all.

Reply:
Done